lib/rest-api.php

```diff
@@ -30,3 +30,51 @@ function gutenberg_register_edit_site_export_controller_endpoints() {
 }
 
 add_action( 'rest_api_init', 'gutenberg_register_edit_site_export_controller_endpoints' );
+
+if ( ! function_exists( 'gutenberg_register_wp_rest_post_types_controller_fields' ) ) {
+	/**
+	 * Adds `template` and `template_lock` fields to WP_REST_Post_Types_Controller class.
+	 */
+	function gutenberg_register_wp_rest_post_types_controller_fields() {
+		register_rest_field(
+			'type',
+			'template',
+			array(
+				'get_callback' => function ( $item ) {
+					$post_type = get_post_type_object( $item['slug'] );
+					if ( ! empty( $post_type ) && ! empty( $post_type->template ) ) {
+						return $post_type->template;
+					}
+					return null;
+				},
+				'schema'       => array(
+					'type'        => 'array',
+					'description' => __( 'The template associated with the post type.', 'gutenberg' ),
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+			)
+		);
+		register_rest_field(
+			'type',
+			'template_lock',
+			array(
+				'get_callback' => function ( $item ) {
+					$post_type = get_post_type_object( $item['slug'] );
+					if ( ! empty( $post_type ) && ! empty( $post_type->template_lock ) ) {
+						return $post_type->template_lock;
+					}
+					return null;
```

Comment:
According to [this other logic](https://github.com/WordPress/wordpress-develop/blob/5c7f94a6aa979dc70ff935e9006957fe0f262932/src/wp-admin/edit-form-blocks.php#L231), it should return `false`.

Reply:
Hi @oandregal, setting a value of false is the same as not having template_lock, I guess we can ignore the false values, and return null/not return when locking is not set either because there is no locking referred at all or because it is false. I don't see any downside to that and this would allow us to use an enum as a type with only array( 'all', 'insert', 'contentOnly' ). We can not really have a type that is enum or boolean. What do you think?

Reply:
What about nested content? Is there a difference between `false` or `null` in that scenario? I'm thinking about the locking lookup: if I remember this correctly, if a block (`wp_navigation` or `wp_block` types, for example) doesn't have a lock, we look it up above in the parents. I don't remember the details of "does not have a lock": was that checking for `false` or for `null`/non existence (or both)?

Other than double checking that scenario, the current approach makes sense to me.

Reply:
The nested locking is something at the block level, this template lock is only used now on edit-post, and the template locks of CPTs can not be nested. The locking of templates on CPTs is always top level there is never a need to overite with false as may happen on the block level. So I think we are safe not returning the false value at all.

Reply:
Ah, that sounds good. Thanks for the braindump.